airflow/providers/amazon/aws/example_dags/example_glue.py

```diff
@@ -0,0 +1,124 @@
+GLUE_CRAWLER_ROLE = getenv('GLUE_CRAWLER_ROLE', 'glue_crawler_role')
+GLUE_CRAWLER_NAME = 'example_crawler'
+GLUE_CRAWLER_CONFIG = {
+    'Name': GLUE_CRAWLER_NAME,
+    'Role': GLUE_CRAWLER_ROLE,
+    'DatabaseName': GLUE_DATABASE_NAME,
+    'Targets': {
+        'S3Targets': [
+            {
+                'Path': f'{GLUE_EXAMPLE_S3_BUCKET}/input',
+            }
+        ]
+    },
+}
+
+# Example csv data used as input to the example AWS Glue Job.
+EXAMPLE_CSV = '''
+food,price
+apple,0.5
+milk,2.5
+bread,4.0
+'''
+
+# Example Spark script to operate on the above sample csv data.
+EXAMPLE_SCRIPT = f'''
+from pyspark.context import SparkContext
+from awsglue.context import GlueContext
+
+glueContext = GlueContext(SparkContext.getOrCreate())
+datasource = glueContext.create_dynamic_frame.from_catalog(
+             database='{GLUE_DATABASE_NAME}', table_name='input')
+print('There are %s items in the table' % datasource.count())
+
+datasource.toDF().write.format('csv').mode("append").save('s3://{GLUE_EXAMPLE_S3_BUCKET}/output')
+'''
+
+
+@task(task_id='setup__upload_artifacts_to_s3')
+def upload_artifacts_to_s3():
+    '''Upload example CSV input data and an example Spark script to be used by the Glue Job'''
+    s3_hook = S3Hook()
+    s3_load_kwargs = {"replace": True, "bucket_name": GLUE_EXAMPLE_S3_BUCKET}
+    s3_hook.load_string(string_data=EXAMPLE_CSV, key='input/input.csv', **s3_load_kwargs)
+    s3_hook.load_string(string_data=EXAMPLE_SCRIPT, key='etl_script.py', **s3_load_kwargs)
+
+
+with DAG(
+    dag_id='example_glue',
+    schedule_interval=None,
+    start_date=datetime(2021, 1, 1),
+    tags=['example'],
+    catchup=False,
+) as glue_dag:
+
+    setup_upload_artifacts_to_s3 = upload_artifacts_to_s3()
+
+    # [START howto_operator_glue_crawler]
+    crawl_s3 = GlueCrawlerOperator(
+        task_id='crawl_s3',
+        config=GLUE_CRAWLER_CONFIG,
+        wait_for_completion=False,
+    )
+    # [END howto_operator_glue_crawler]
+
+    # [START howto_sensor_glue_crawler]
+    wait_for_crawl = GlueCrawlerSensor(task_id='wait_for_crawl', crawler_name=GLUE_CRAWLER_NAME)
+    # [END howto_sensor_glue_crawler]
+
+    # [START howto_operator_glue]
+    job_name = 'example_glue_job'
+    submit_glue_job = GlueJobOperator(
+        task_id='submit_glue_job',
+        job_name=job_name,
+        wait_for_completion=False,
+        script_location=f's3://{GLUE_EXAMPLE_S3_BUCKET}/etl_script.py',
+        s3_bucket=GLUE_EXAMPLE_S3_BUCKET,
+        iam_role_name=GLUE_CRAWLER_ROLE.split('/')[-1],
+        create_job_kwargs={'GlueVersion': '3.0', 'NumberOfWorkers': 2, 'WorkerType': 'G.1X'},
+    )
+    # [END howto_operator_glue]
+
+    # [START howto_sensor_glue]
+    wait_for_job = GlueJobSensor(
+        task_id='wait_for_job',
+        job_name=job_name,
+        # Job ID extracted from previous Glue Job Operator task
+        run_id="{{ ti.xcom_pull(task_ids='submit_glue_job') }}",
```

Comment:
```suggestion
        run_id=submit_glue_job.output,
```
Another perk of the TaskFlow API: using the `.output` property or [`XComArgs`](https://airflow.apache.org/docs/apache-airflow/stable/tutorial_taskflow_api.html#consuming-xcoms-between-decorated-and-traditional-tasks). This is a functional abstraction over the classic `"{{ ti.xcom_pull(...) }}"` approach to pull `XComs` from operators.

Reply:
Sure, updated the PR with this suggestion :)

Note: I also removed the header from the example CSV file since it was causing issues in the glue job, so when you see that in the diff, it was intentional :+1: